Improve documentation for ACME/Let's Encrypt (#5819)
containous/traefik: docs/content/https/acme.md

!!! warning "Let's Encrypt and Rate Limiting"
    Note that Let's Encrypt API has [rate limiting](https://letsencrypt.org/docs/rate-limits).

>     Use Let's Encrypt staging server with the [`caServer`](#caserver) configuration directives

Reviewer: ```suggestion
    Use Let's Encrypt staging server with the [`caServer`](#caserver) configuration option
```

Author: Challenging this suggestion: looks fine, but how do we deal with the potential confusion with `tls.options` option? I initially wrote option as you suggests, but felt that `directives` was a mor non-ambiguous wording in this context (TLS). WDYT?

Reviewer 2: We will keep `option` to be homogeneous with the other documentation

    when experimenting to avoid hitting this limit too fast.
    
## Certificate Resolvers

Traefik requires you to define "Certificate Resolvers" in the [static configuration](../getting-started/configuration-overview.md#the-static-configuration), 
which are responsible for retrieving certificates from an ACME server.

Then, each ["router"](../routing/routers/index.md) is configured to enable TLS,

> and is associated to a certificate resolver through the [`tls.certresolver` configuration directive](../routing/routers/index.md#certresolver).

Reviewer: ```suggestion
and is associated to a certificate resolver through the [`tls.certresolver` configuration option](../routing/routers/index.md#certresolver).
```

Author: Challenging this suggestion: looks fine, but how do we deal with the potential confusion with `tls.options` option? I initially wrote option as you suggests, but felt that `directives` was a mor non-ambiguous wording in this context (TLS). WDYT?

Reviewer 2: We will keep `option` to be homogeneous with the other documentation

Certificates are requested for domain names retrieved from the router's [dynamic configuration](../getting-started/configuration-overview.md#the-dynamic-configuration).

> You can read more about this retrieval mechanism on the following section: [ACME Domain Definition](#domain-definition).

Reviewer: ```suggestion
You can read more about this retrieval mechanism in the following section: [ACME Domain Definition](#domain-definition).
```

## Domain Definition

Certificate resolvers request certificates for a set of the domain names 
inferred from routers, with the following logic:

> - If the router has a directive [`tls.domains`](../routing/routers/index.md#domains) set,

Reviewer: ```suggestion
- If the router has a [`tls.domains`](../routing/routers/index.md#domains) option set,
```

Author: Challenging this suggestion: looks fine, but how do we deal with the potential confusion with `tls.options` option? I initially wrote option as you suggests, but felt that `directives` was a mor non-ambiguous wording in this context (TLS). WDYT?

Reviewer 2: We will keep `option` to be homogeneous with the other documenation

>   then the certificate resolver uses the sub-directives `main` (and optionnaly `sans`) of `tls.domains` to know the domain names for this router.

Reviewer: ```suggestion
  then the certificate resolver uses the `main` (and optionally `sans`) option of `tls.domains` to know the domain names for this router.
```

Author: Challenging this suggestion: looks fine, but how do we deal with the potential confusion with `tls.options` option? I initially wrote option as you suggests, but felt that `directives` was a mor non-ambiguous wording in this context (TLS). WDYT? 

[Even though I'm adding the suggestion because it fixes a typo)

> - If no [`tls.domains`](../routing/routers/index.md#domains) directive is set, 

Reviewer: ```suggestion
- If no [`tls.domains`](../routing/routers/index.md#domains) option is set, 
```

Author: Challenging this suggestion: looks fine, but how do we deal with the potential confusion with `tls.options` option? I initially wrote option as you suggests, but felt that `directives` was a mor non-ambiguous wording in this context (TLS). WDYT?

Reviewer 2: We will keep `option` to be homogeneous with the other documentation

  then the certificate resolver uses the [router's rule](../routing/routers/index.md#rule), 
  by checking the `Host()` matchers. 
  Please note that [multiple `Host()` matchers can be used](../routing/routers/index.md#certresolver)) for specifying multiple domain names for this router.

> Please note that that:

Reviewer: ```suggestion
Please note that:
```
